smartsim/entity/model.py

```diff
@@ -258,11 +258,12 @@ def colocate_db_uds(
                 f"Invalid name for unix socket: {unix_socket}. Must only "
                 "contain alphanumeric characters or . : _ - /"
             )
-
-        uds_options = {
+        assert isinstance(unix_socket, str)
+        assert isinstance(socket_permissions, int)
```

Comment:
Can we remove the `assert` stmts here? It looks like mypy should already be able to eventuate these types correctly